src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INCREASE = 2;
+    private int currentCapacity;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
+        increaseCapacity();
 
+        Node<K, V> node = new Node<>(key, value, null);
+        int index = getBucketIndex(key);
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = node;
+                size++;
+            }
+            currentNode = currentNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getBucketIndex(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        int hash;
+        return (key == null) ? 0 : Math.abs((hash = key.hashCode()) ^ (hash >>> 16));
+    }
+
+    private int getBucketIndex(K key) {
+        if (getHash(key) == 0) {
+            return 0;
+        }
+        return Math.abs(getHash(key) % table.length);
+    }
```

Comment:
Do you understand how it works? It looks redundant. You can use only gatBucketIndex Method. key has its own hashCode method.
```suggestion
       private int getBucketIndex(K key) {
        if (key == null) {
            return 0;
        }
        return Math.abs(key.hashCode() % table.length);
    }
```

Reply:
yes, i know how it works:)